### PR TITLE
DOC-426: Add initial MyRocks documentation

### PR DESCRIPTION
**Pull request by azhebel**

### CODE REVIEW

File: doc/source/myrocks/install.rst [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+Finally, remove all the :ref:`myrocks_server_variables`
+from the configuration file (:file:`my.cnf`)
+and restart Percona Server.
+
+Migrating from Facebook MyRocks
+===============================
+
+Do we need this? Is the process similar to migrating from InnoDB?
+https://github.com/facebook/mysql-5.6/wiki/Migrating-from-InnoDB-to-RocksDB
+
```
Comment (georgelorchpercona) [inline doc/source/myrocks/install.rst]: I would not get into migrating from FB MyRocks or MariaDB MyRocks or any other 'variant' as we have made it clear already above that we do not support anything other than logical dump/load to Percona MyRocks.

**Comment by georgelorchpercona** [comment]
Another thing, this link should list items in JIRA that might be worth making some explicit notes in doc as they are point where we differ from upstream Facebook MyRocks https://jira.percona.com/issues/?filter=12007&jql=project%20%3D%20MYR%20AND%20status%20%3D%20Done%20AND%20labels%20in%20(docs%2C%20documentation)%20ORDER%20BY%20created%20ASC

A couple of the bigger things that are probably worth calling out:
https://jira.percona.com/browse/MYR-28 : Support lz4 and zstd compression by static link
https://jira.percona.com/browse/MYR-35 : Alter gap lock error logic to account for ISO mode
https://jira.percona.com/browse/MYR-124 : START TRANSACTION WITH CONSISTENT SNAPSHOT behaves different

File: doc/source/myrocks/install.rst (hunk 1 above)
Comment (azhebel) [inline doc/source/myrocks/install.rst]: @georgelorchpercona I was thinking that we could describe exactly that: the dump/load procedure for those who are already using FB MyRocks (or possibly MariaDB). Possibly also describe how to migrate from InnoDB/XtraDB. But maybe this is something for the GA release. We don't want to encourage anyone migrating their production workloads for now, right?

**Comment by azhebel** [comment]
@georgelorchpercona the server and status variables were taken from upstream by Hrvoje as he started documenting MyRocks some time ago. This is a very big work as I will need to go through all of them, add info and descriptions. I will gradually finalize it as we get closer to GA. If you have a list of vars in Percona MyRocks (or which ones are omitted), please let me know. Otherwise I guess I will need to just go through our source code and see which ones are there.

**Comment by georgelorchpercona** [comment]
Honestly, I didn't expect any of this for the experimental release, but certainly for GA. The thing is though, with me continually merging upstream in big chunks, these are a moving target as they keep adding new ones and unfortunately occasionally renaming some. Once we GA we will not be able to accept any renames.

I can get you accurate listing of what we actually imported (or you can if you have a running instance). There is no need for you to have to root through the source to find out.

**Comment by azhebel** [comment]
I do have a running instance and I can run SHOW VARIABLES and SHOW STATUS. This way I can also know which are global/session-only. I can also use descriptions from FB and code (https://github.com/percona/percona-server/blob/5.7/storage/rocksdb/ha_rocksdb.cc). How do I easily know which variables have a corresponding command-line option? Also, which are dynamic?

**Comment by georgelorchpercona** [comment]
Hmm, that is a bit more tricky because I do not believe that can be represented via SQL in any way. Maybe you will have to look at the code anyway. You can look at storage/rocksdb/ha_rocksdb.cc for all of the definitions of the global and session variables. Globals are defined with MYSQL_SYSVAR_<type> and sessions are defined with MYSQL_THDVAR_<type>

You can extract brief descriptions, min, max, and default values from these macros but you kind of need to know how/where to look as there are differences between types and scopes in the positions where they show in the macro. I can help with this and Hrvoje has some experience doing this as well and may have some tip or trick for you.

File: doc/source/myrocks/install.rst [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+.. contents::
+   :local:
+
+Installing Percona MyRocks
+==========================
+
+Percona provides packages for most popular 64-bit Linux distributions:
+
+* Debian 7 ("wheezy")
+* Debian 8 ("jessie")
```
Comment (hrvojem) [inline doc/source/myrocks/install.rst]: AFAIK there is no Debian 7 support

File: doc/source/myrocks/install.rst [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+* Ubuntu 16.04 LTS (Xenial Xerus)
+* Ubuntu 16.10 (Yakkety Yak)
+* Ubuntu 17.04 (Zesty Zapus)
+* Red Hat Enterprise Linux or CentOS 6 (Santiago)
+* Red Hat Enterprise Linux or CentOS 7 (Maipo)
+
+.. note:: Percona Toolkit should work on other DEB-based and RPM-based systems
+   (for example, Oracle Linux and Amazon Linux AMI),
+   but it is tested only on those listed above.
+
```
Comment (hrvojem) [inline doc/source/myrocks/install.rst]: PT mention

File: doc/source/myrocks/install.rst (hunk 1 above)
Comment (hrvojem) [inline doc/source/myrocks/install.rst]: @georgelorchpercona I pulled the list of variables from version that's currently built for testing (in the experimental repo).

File: doc/source/myrocks/limitations.rst [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+* `Transportable tablespace <https://dev.mysql.com/doc/refman/5.7/en/innodb-transportable-tablespace-examples.html>`_
+
+* `Foreign keys <https://dev.mysql.com/doc/refman/5.7/en/create-table-foreign-keys.html>`_
+
+* `Spatial indexes <https://dev.mysql.com/doc/refman/5.7/en/using-spatial-indexes.html>`_
+
+* `Fulltext indexes <https://dev.mysql.com/doc/refman/5.7/en/innodb-fulltext-index.html>`_
+
+* `Gap locks <https://dev.mysql.com/doc/refman/5.7/en/innodb-locking.html#innodb-gap-locks>`_
+
```
Comment (hrvojem) [inline doc/source/myrocks/limitations.rst]: add the following per @georgelorchpercona suggestion:

> MyRocks does not support operating as either a master or a slave in any replication topology that is not exclusively ROW based. No STMT or MIX.

**Pull request merged**